### DISCUSSION TITLE
Signing with certificates

### DISCUSSION THREAD

**Discussion by Softbox44** [Stirling-Tools/Stirling-PDF#941] [Q&A]
Is there any documentation on howto generate abd set certificates up. And what the difference is between the 3 certificate types.
thanks!

**Comment by sbplat** [comment]
You can generate keys and certificates (self-signed) using `openssl`. The different types are the formats we directly support, so for other types, you will need to convert them using something like `keytool`.

**Comment by Softbox44** [comment]
Thanks I will look into it. Openssl usually works with key pairs. How does
that integrate with Sterling? I upload thr public key I guess. And what
would the purpose be? Would the pdf file be encrypted?

**Comment by sbplat** [comment]
1. Yes, you would upload the public key. You should be using a registered certificate issued by a certificate authority as a self-signed one cannot be validated.
2. The purpose is for recipients to be able to verify the legitimacy of the document.
3. No, this feature does not add a password to the PDF. Use add password for that.

**Comment by Softbox44** [comment]
So do I need to send my public key to the recipient as well so they can
verify authenticity?

**Reply by sbplat** [reply]
If it's a self-signed one then they would need to trust your certificate.

**Reply by Frooodle** [reply]
But yes this is correct


Sign with private, verify with public 

So can send public to other people to verify the pdf came from you

**Comment by Softbox44** [comment]
So the certificate is not used to encrypt any data and if its self signed
it also doesn't provide trust. What is the functionality used for then? I
am just trying to understand what i can achieve by setting it up.

**Reply by sbplat** [reply]
There's not much use for self-signed certs. They're mainly used internally such as for testing locally (shouldn't be used in production envs).

You can read more about certificate signatures here: https://helpx.adobe.com/ca/acrobat/kb/certificate-signatures.html
